streams/test-utils/src/test/java/org/apache/kafka/streams/test/TestRecordTest.java

```diff
@@ -166,6 +166,9 @@ public void testProducerRecord() {
         final TestRecord<String, Integer> testRecord = new TestRecord<>(producerRecord);
         final TestRecord<String, Integer> expectedRecord = new TestRecord<>(key, value, headers, recordTime);
         assertEquals(expectedRecord, testRecord);
-        assertNotEquals(expectedRecord, producerRecord);
+        assertFalse(expectedRecord.getHeaders() == producerRecord.headers() &&
```

Comment:
Also, it should make sure the input arguments (from ```ProducerRecord```) are equals to getter of ```TestRecord```. For example:
```java
assertEquals(expectedRecord.getHeaders(), producerRecord.headers())
```

Reply:
@chia7712 Sorry for the confusion.  Since `expectedRecord` is logically identical to `testRecord` and `producerRecord` is created fro `testRecord`, `expectedRecord` and `testRecord` are logically equal. The reason why the previous assertion was always true was not for its logical state, but the difference between their classes - `ProducerRecord` vs. `TestRecord`.

So, after some thought, I concluded that just removing the meaningless assertion is enough. Here is the update.

Reply:
fair enough :)

Reply:
Is the assert  intentional? If so, we should keep it and then add clear comments to avoid confusing readers in the future.

WDYT?

Reply:
> Is the assert intentional?

It seems not. If it was interntional, the same statement must be included in its consumer-side counterpart test method, `testConsumerRecord`, which is actually not.